File: mobile-test/mobile-test/Address.swift
//
//  Address.swift
//  mobile-test
//
//  Created by Martin Gyupchanov on 09-08-17.
//  Copyright © 2017 Martin Gyupchanov. All rights reserved.
//

import Foundation

struct Address {
  var address1: String?
  var label: String?
  var zipCode: String?
  var city: String?
  var state: String?
  var country: String?
  var latitude: String?
  var longitude: String?
  
  var allValidProperties = [String]()
  
  init(address1: String?, label: String?, zipCode: String?, city: String?, state: String?, country: String?, latitude: String?, longitude: String?) {
    self.address1 = address1
    self.label = label
    self.zipCode = zipCode
    self.city = city
    self.state = address1
    self.country = country
    self.latitude = latitude
    self.longitude = longitude

    if let address1 = address1 {
      allValidProperties.append(address1)
    }
    
    if let label = label {
      allValidProperties.append(label)
    }
    
    if let zipCode = zipCode {
      allValidProperties.append(zipCode)
    }
    
    if let city = city {
      allValidProperties.append(city)
    }
    
    if let state = state {
      allValidProperties.append(state)
    }
    
    if let country = country {
      allValidProperties.append(country)
    }
    
    if let latitude = latitude {
      allValidProperties.append(latitude)
    }
    
    if let longitude = longitude {
      allValidProperties.append(longitude)
    }
  }
}

File: QuickSeries/app/src/main/java/com/test/quickseries/base/model/Address.java
package com.test.quickseries.base.model;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.SerializedName;

/**
 * Created by elton on 2017-10-14.
 */

public class Address implements Parcelable {

    @SerializedName("address1")
    private String address1;
    @SerializedName("label")
    private String label;
    @SerializedName("zipCode")
    private String zipCode;
    @SerializedName("city")
    private String city;
    @SerializedName("state")
    private String state;
    @SerializedName("country")
    private String country;
    @SerializedName("gps")
    private Gps gps;

    public String getAddress1() {
        return address1;
    }

    public void setAddress1(String address1) {
        this.address1 = address1;
    }

    public String getLabel() {
        return label;
    }

    public void setLabel(String label) {
        this.label = label;
    }

    public String getZipCode() {
        return zipCode;
    }

    public void setZipCode(String zipCode) {
        this.zipCode = zipCode;
    }

    public String getCity() {
        return city;
    }

    public void setCity(String city) {
        this.city = city;
    }

    public String getState() {
        return state;
    }

    public void setState(String state) {
        this.state = state;
    }

    public String getCountry() {
        return country;
    }

    public void setCountry(String country) {
        this.country = country;
    }

    public Gps getGps() {
        return gps;
    }

    public void setGps(Gps gps) {
        this.gps = gps;
    }


    protected Address(Parcel in) {
        address1 = in.readString();
        label = in.readString();
        zipCode = in.readString();
        city = in.readString();
        state = in.readString();
        country = in.readString();
        gps = (Gps) in.readValue(Gps.class.getClassLoader());
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(address1);
        dest.writeString(label);
        dest.writeString(zipCode);
        dest.writeString(city);
        dest.writeString(state);
        dest.writeString(country);
        dest.writeValue(gps);
    }

    @SuppressWarnings("unused")
    public static final Parcelable.Creator<Address> CREATOR = new Parcelable.Creator<Address>() {
        @Override
        public Address createFromParcel(Parcel in) {
            return new Address(in);
        }

        @Override
        public Address[] newArray(int size) {
            return new Address[size];
        }
    };

}

